test/__tests__/stdin-filepath.js

```diff
@@ -0,0 +1,151 @@
+  ).test({
+    status: 0,
+  });
+});
+
+describe("throw error if stdin content incompatible with stdin-filepath", () => {
+  runCli(
+    "",
+    ["--stdin-filepath", "abc.js"],
+    { input: ".name { display: none; }" }, // css
+  ).test({
+    status: "non-zero",
+  });
+});
+
+describe("gracefully handle stdin-filepath with nonexistent directory", () => {
+  runCli(
+    "",
+    ["--stdin-filepath", "definitely/nonexistent/path.css"],
+    { input: ".name { display: none; }" }, // css
+  ).test({
+    status: 0,
+  });
+});
+
+describe("apply editorconfig for stdin-filepath with nonexistent file", () => {
+  runCli("editorconfig", ["--stdin-filepath", "nonexistent.js"], {
+    input: dedent`
+      function f() {
+        console.log("should be indented with a tab");
+      }
+    `, // js
+  }).test({
+    status: 0,
+  });
+});
+
+describe("apply editorconfig for stdin-filepath with nonexistent directory", () => {
+  runCli(
+    "editorconfig",
+    ["--stdin-filepath", "nonexistent/one/two/three.js"],
+    {
+      input: dedent`
+        function f() {
+          console.log("should be indented with a tab");
+        }
+      `, // js
+    },
+  ).test({
+    status: 0,
+  });
+});
+
+describe("apply editorconfig for stdin-filepath with a deep path", () => {
+  runCli(
+    "editorconfig",
+    ["--stdin-filepath", "a/".repeat(30) + "three.js"],
+    {
+      input: dedent`
+        function f() {
+          console.log("should be indented with a tab");
+        }
+      `, // js
+    },
+  ).test({
+    status: 0,
+  });
+});
+
+// TODO: This is currently a false positive as no config actually gets resolved, but Prettier
+// somehow formats the input correctly anyway.
+describe("apply editorconfig for stdin-filepath in root", () => {
+  const code = dedent`
+    function f() {
+      console.log("should be indented with a tab");
+    }
+  `;
+  runCli("", ["--stdin-filepath", "/foo.js"], {
+    input: code, // js
+  }).test({
+    status: 0,
+    stdout: code,
+    stderr: "",
+    write: [],
+  });
+});
+
+describe("apply editorconfig for stdin-filepath with a deep path", () => {
+  runCli(
+    "editorconfig",
+    ["--stdin-filepath", "a/".repeat(30) + "three.js"],
+    {
+      input: dedent`
+        function f() {
+          console.log("should be indented with a tab");
+        }
+      `, // js
+    },
+  ).test({
+    status: 0,
+  });
+});
+
+// TODO: This is currently a false positive. Gotta investigate how it's handled in Prettier v3 to
+// gauge the expected behavior.
+describe("don't apply editorconfig outside project for stdin-filepath with nonexistent directory", () => {
+  runCli(
+    "",
+    [
+      "--stdin-filepath",
+      "editorconfig/repo-root/nonexistent/one/two/three.js",
+    ],
+    {
+      input: dedent`
+        function f() {
+          console.log("should be indented with 2 spaces");
+        }
+      `, // js
+    },
+  ).test({
+    status: 0,
+  });
+});
```

Comment:
It's currently a false positive, but when we change the first argument in `runCli` to `editorconfig` and update the path accordingly, `stdin` does get formatted based on `__fixtures__/editorconfig/.editorconfig`. Not sure why it wouldn't based on other test cases. Is there some special case for `repo-root` directories?